packages/core/src/main/notification/NotificationService.ts

```diff
@@ -453,4 +443,78 @@ export class NotificationService extends EventEmitter {
       this.logger.error('Could not get pending proposals', error);
     }
   }
+
+  /**
+   * ## MLS only ##
+   * Store groupIds with last key material update dates.
+   *
+   * @param {groupId} params.groupId - groupId of the mls conversation
+   * @param {previousUpdateDate} params.previousUpdateDate - date of the previous key material update
+   */
+  public async storeLastKeyMaterialUpdateDate(params: LastKeyMaterialUpdateParams) {
+    await this.database.storeLastKeyMaterialUpdateDate(params);
+    await this.scheduleTaskToRenewKeyMaterial(params);
+  }
+
+  /**
+   * ## MLS only ##
+   * Renew key material for a given groupId
+   *
+   * @param groupId groupId of the conversation
+   */
+  private async renewKeyMaterial({groupId}: Omit<LastKeyMaterialUpdateParams, 'previousUpdateDate'>) {
+    try {
+      const groupIdDecodedFromBase64 = Decoder.fromBase64(groupId).asBytes;
+      const commitBundle = await this.mlsService.updateKeyingMaterial(groupIdDecodedFromBase64);
+      await this.mlsService.uploadCoreCryptoCommitBundle(groupIdDecodedFromBase64, commitBundle);
+
+      const keyRenewalTime = new Date().getTime();
+      const keyMaterialUpdateDate = {groupId, previousUpdateDate: keyRenewalTime};
+      await this.storeLastKeyMaterialUpdateDate(keyMaterialUpdateDate);
+    } catch (error) {
+      this.logger.error(`Error while renewing key material for groupId ${groupId}`, error);
+    }
+  }
+
+  private createKeyMaterialUpdateTaskSchedulerId(groupId: string) {
+    return `renew_key_material-${groupId}`;
+  }
+
+  private async scheduleTaskToRenewKeyMaterial({groupId, previousUpdateDate}: LastKeyMaterialUpdateParams) {
+    const groupConversationExists = await this.mlsService.conversationExists(Decoder.fromBase64(groupId).asBytes);
+
+    if (!groupConversationExists) {
```

Comment:
If for some reason user is no longer a member of the group we store `last-key-material-update` date for, we need to remove its entry and don't schedule a renewal task.